app/database/group_item_store_compute_all_access.go

```diff
@@ -112,14 +113,12 @@ func (s *GroupItemStore) computeAllAccess() {
 			FROM groups_items AS child
 			JOIN items_items
 				ON items_items.child_item_id = child.item_id
-			LEFT JOIN groups_items_propagate
-				ON groups_items_propagate.id = child.id
 			JOIN groups_items AS parent
 				ON parent.item_id = items_items.parent_item_id AND parent.group_id = child.group_id
 			JOIN items AS parent_item
 				ON parent_item.id = items_items.parent_item_id
 			WHERE
-				(groups_items_propagate.propagate_access = 'self' OR groups_items_propagate.id IS NULL) AND
+				child.id = groups_items_propagate.id AND
```

Comment:
Is it useful? The "using (id)" should do the job correctly, but it depends if it succeeds doing it in an efficient way.

Reply:
It was added to optimize the query. Otherwise MySQL aggregated info for all the groups_items.

Reply:
With filtering:
```
mysql> explain UPDATE groups_items JOIN groups_items_propagate USING(id) LEFT JOIN LATERAL ( SELECT child.id, MIN(parent.cached_full_access_date) AS cached_full_access_date, MIN(IF(items_items.partial_access_propagation = 'AsPartial', parent.cached_partial_access_date, NULL)) AS cached_partial_access_date, MAX(parent.cached_manager_access) AS cached_manager_access, MIN(IF(items_items.partial_access_propagation = 'AsGrayed', parent.cached_partial_access_date, NULL)) AS cached_grayed_access_date, MIN(parent.cached_access_solutions_date) AS cached_access_solutions_date, CONCAT('From ancestor group(s) ', GROUP_CONCAT( DISTINCT IF(parent.access_reason = '', NULL, parent.access_reason) ORDER BY parent_item.id SEPARATOR ', ' )) AS access_reason_ancestors FROM groups_items AS child JOIN items_items ON items_items.child_item_id = child.item_id JOIN groups_items AS parent ON parent.item_id = items_items.parent_item_id AND parent.group_id = child.group_id JOIN items AS parent_item ON parent_item.id = items_items.parent_item_id WHERE child.id = groups_items_propagate.id AND ( parent.cached_full_access_date IS NOT NULL OR (parent.cached_partial_access_date IS NOT NULL AND items_items.partial_access_propagation != 'None') OR parent.cached_access_solutions_date IS NOT NULL OR parent.cached_manager_access ) AND parent_item.custom_chapter = 0 GROUP BY child.id ) AS new_data USING(id) SET groups_items.cached_full_access_date = LEAST( IFNULL(new_data.cached_full_access_date, groups_items.full_access_date), IFNULL(groups_items.full_access_date, new_data.cached_full_access_date) ), groups_items.cached_partial_access_date = LEAST( IFNULL(new_data.cached_partial_access_date, groups_items.partial_access_date), IFNULL(groups_items.partial_access_date, new_data.cached_partial_access_date) ), groups_items.cached_manager_access = GREATEST( IFNULL(new_data.cached_manager_access, 0), groups_items.manager_access ), groups_items.cached_access_solutions_date = LEAST( IFNULL(new_data.cached_access_solutions_date, groups_items.access_solutions_date), IFNULL(groups_items.access_solutions_date, new_data.cached_access_solutions_date) ), groups_items.cached_grayed_access_date = new_data.cached_grayed_access_date, groups_items.cached_access_reason = new_data.access_reason_ancestors;
+----+-------------------+------------------------+------------+--------+-----------------------------------------------------------------------------------------+------------------+---------+---------------------------------------------------------------------------+--------+----------+-----------------------------------------+
| id | select_type       | table                  | partitions | type   | possible_keys                                                                           | key              | key_len | ref                                                                       | rows   | filtered | Extra                                   |
+----+-------------------+------------------------+------------+--------+-----------------------------------------------------------------------------------------+------------------+---------+---------------------------------------------------------------------------+--------+----------+-----------------------------------------+
|  1 | PRIMARY           | groups_items_propagate | NULL       | index  | PRIMARY                                                                                 | propagate_access | 1       | NULL                                                                      | 200191 |   100.00 | Using index; Rematerialize (<derived2>) |
|  1 | UPDATE            | groups_items           | NULL       | eq_ref | PRIMARY                                                                                 | PRIMARY          | 8       | algorea_example.groups_items_propagate.id                                 |      1 |   100.00 | NULL                                    |
|  1 | PRIMARY           | <derived2>             | NULL       | ref    | <auto_key0>                                                                             | <auto_key0>      | 8       | algorea_example.groups_items_propagate.id                                 |      2 |   100.00 | NULL                                    |
|  2 | DEPENDENT DERIVED | child                  | NULL       | eq_ref | PRIMARY,item_id,version,group_id,itemtem_id,full_access,access_solutions,partial_access | PRIMARY          | 8       | algorea_example.groups_items_propagate.id                                 |      1 |   100.00 | NULL                                    |
|  2 | DEPENDENT DERIVED | items_items            | NULL       | ref    | parent_item_id,child_item_id,parent_child,parent_version                                | child_item_id    | 8       | algorea_example.child.item_id                                             |      1 |   100.00 | NULL                                    |
|  2 | DEPENDENT DERIVED | parent                 | NULL       | eq_ref | item_id,group_id,itemtem_id                                                             | item_id          | 16      | algorea_example.items_items.parent_item_id,algorea_example.child.group_id |      1 |    99.99 | Using where                             |
|  2 | DEPENDENT DERIVED | parent_item            | NULL       | eq_ref | PRIMARY                                                                                 | PRIMARY          | 8       | algorea_example.items_items.parent_item_id                                |      1 |    10.00 | Using where                             |
+----+-------------------+------------------------+------------+--------+-----------------------------------------------------------------------------------------+------------------+---------+---------------------------------------------------------------------------+--------+----------+-----------------------------------------+
7 rows in set, 2 warnings (0.02 sec)
```
Without filtering:
```
mysql> explain UPDATE groups_items JOIN groups_items_propagate USING(id) LEFT JOIN ( SELECT child.id, MIN(parent.cached_full_access_date) AS cached_full_access_date, MIN(IF(items_items.partial_access_propagation = 'AsPartial', parent.cached_partial_access_date, NULL)) AS cached_partial_access_date, MAX(parent.cached_manager_access) AS cached_manager_access, MIN(IF(items_items.partial_access_propagation = 'AsGrayed', parent.cached_partial_access_date, NULL)) AS cached_grayed_access_date, MIN(parent.cached_access_solutions_date) AS cached_access_solutions_date, CONCAT('From ancestor group(s) ', GROUP_CONCAT( DISTINCT IF(parent.access_reason = '', NULL, parent.access_reason) ORDER BY parent_item.id SEPARATOR ', ' )) AS access_reason_ancestors FROM groups_items AS child JOIN items_items ON items_items.child_item_id = child.item_id JOIN groups_items AS parent ON parent.item_id = items_items.parent_item_id AND parent.group_id = child.group_id JOIN items AS parent_item ON parent_item.id = items_items.parent_item_id WHERE ( parent.cached_full_access_date IS NOT NULL OR (parent.cached_partial_access_date IS NOT NULL AND items_items.partial_access_propagation != 'None') OR parent.cached_access_solutions_date IS NOT NULL OR parent.cached_manager_access ) AND parent_item.custom_chapter = 0 GROUP BY child.id ) AS new_data USING(id) SET groups_items.cached_full_access_date = LEAST( IFNULL(new_data.cached_full_access_date, groups_items.full_access_date), IFNULL(groups_items.full_access_date, new_data.cached_full_access_date) ), groups_items.cached_partial_access_date = LEAST( IFNULL(new_data.cached_partial_access_date, groups_items.partial_access_date), IFNULL(groups_items.partial_access_date, new_data.cached_partial_access_date) ), groups_items.cached_manager_access = GREATEST( IFNULL(new_data.cached_manager_access, 0), groups_items.manager_access ), groups_items.cached_access_solutions_date = LEAST( IFNULL(new_data.cached_access_solutions_date, groups_items.access_solutions_date), IFNULL(groups_items.access_solutions_date, new_data.cached_access_solutions_date) ), groups_items.cached_grayed_access_date = new_data.cached_grayed_access_date, groups_items.cached_access_reason = new_data.access_reason_ancestors;
+----+-------------+------------------------+------------+--------+-----------------------------------------------------------------------------------------+------------------+---------+---------------------------------------------------------------------------+--------+----------+-------------+
| id | select_type | table                  | partitions | type   | possible_keys                                                                           | key              | key_len | ref                                                                       | rows   | filtered | Extra       |
+----+-------------+------------------------+------------+--------+-----------------------------------------------------------------------------------------+------------------+---------+---------------------------------------------------------------------------+--------+----------+-------------+
|  1 | PRIMARY     | groups_items_propagate | NULL       | index  | PRIMARY                                                                                 | propagate_access | 1       | NULL                                                                      | 200191 |   100.00 | Using index |
|  1 | UPDATE      | groups_items           | NULL       | eq_ref | PRIMARY                                                                                 | PRIMARY          | 8       | algorea_example.groups_items_propagate.id                                 |      1 |   100.00 | NULL        |
|  1 | PRIMARY     | <derived2>             | NULL       | ref    | <auto_key0>                                                                             | <auto_key0>      | 8       | algorea_example.groups_items_propagate.id                                 |     10 |   100.00 | NULL        |
|  2 | DERIVED     | child                  | NULL       | index  | PRIMARY,item_id,version,group_id,itemtem_id,full_access,access_solutions,partial_access | PRIMARY          | 8       | NULL                                                                      | 205815 |   100.00 | NULL        |
|  2 | DERIVED     | items_items            | NULL       | ref    | parent_item_id,child_item_id,parent_child,parent_version                                | child_item_id    | 8       | algorea_example.child.item_id                                             |      1 |   100.00 | NULL        |
|  2 | DERIVED     | parent                 | NULL       | eq_ref | item_id,group_id,itemtem_id                                                             | item_id          | 16      | algorea_example.items_items.parent_item_id,algorea_example.child.group_id |      1 |    99.99 | Using where |
|  2 | DERIVED     | parent_item            | NULL       | eq_ref | PRIMARY                                                                                 | PRIMARY          | 8       | algorea_example.items_items.parent_item_id                                |      1 |    10.00 | Using where |
+----+-------------+------------------------+------------+--------+-----------------------------------------------------------------------------------------+------------------+---------+---------------------------------------------------------------------------+--------+----------+-------------+
7 rows in set, 1 warning (0.00 sec)```

Reply:
Note the "rows" column